runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -554,12 +554,13 @@ impl<'ctx, 'backend, 'config, C: TxContext, Cfg: Config> StackState<'config>
     }
 
     fn transfer(&mut self, transfer: Transfer) -> Result<(), ExitError> {
+        let mut ctx = self.backend.ctx.borrow_mut();
```

Comment:
Ah I just remembered why we used the "silent" transfer. The ctx you are using is not substate-aware so you cannot use it like this as the substate (EVM's notion of context) containing the internal transfer action may be reverted, but I think your implementation will incorrectly emit an event in this case.

Reply:
Now rebased on top of your https://github.com/oasisprotocol/oasis-sdk/pull/1601 that should resolve this. Thank you!